### PR TITLE
Update homebrew_cask with additional cask features and doc updates

### LINKED ISSUE #1905: homebrew_cask should allow / in name parameter
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

packaging/os/homebrew_cask
##### ANSIBLE VERSION

```
ansible 2.0.1.0
```
##### CONFIGURATION

None
##### OS / ENVIRONMENT

OS X 10.11.4 (15E65) (running against itself on localhost)
##### SUMMARY

When tapping a particular set of Casks, you can install them via tap and the short name, or the full path to the Cask.

e.g.

```
brew cask install fgimian/fgimian/microsoft-office
```

This is also handy when two packages exist with the same name and you wish to install a particular one.

Currently the module does not allow / in the Cask name:

```
# ansible localhost -m homebrew_cask -a "name=fgimian/fgimian/microsoft-office state=present"
localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Invalid cask: fgimian/fgimian/microsoft-office."
}
```
##### STEPS TO REPRODUCE

Simply run the command:

```
ansible localhost -m homebrew_cask -a "name=fgimian/fgimian/microsoft-office state=present"
```

And to demonstrate that this does work with Homebrew Cask, try to install the package manually:

```
brew cask install fgimian/fgimian/microsoft-office
```
##### EXPECTED RESULTS

Package is installed successfully
##### ACTUAL RESULTS

Error is displayed instead (see above)

### PR DESCRIPTION
**Pull request by indrajitr**
##### ISSUE TYPE

- Feature Pull Request (Fixes #1905)
- Docs Pull Request (Syncs up with #1053)
##### COMPONENT NAME

homebrew_cask
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel bced8715cd) last updated 2016/08/04 17:44:21 (GMT -500)
  lib/ansible/modules/core: (devel 95a7ee271a) last updated 2016/08/04 17:44:21 (GMT -500)
  lib/ansible/modules/extras: (stable-2.2 fc76455326) last updated 2016/08/04 17:44:21 (GMT -500)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

- Document missing `path` option and remove redundant brew_path manipulation
- Add 'update_homebrew' option since `brew cask update` is a synonym for
  `brew update` is available nowadays
- Add additional missing aliases documentation
- Port additional improvements (expanded support for brews path, cask path
  patterns etc.)
- Allow proper `list` type support for `name` option.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @indrajitr. This module is going into community review.

We notice that you are one of the maintainers (@indrajitr) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by indrajitr** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @indrajitr. Since you are one of the maintainers (@indrajitr) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
Failing build can be ignored. Missing param docs fixed.

File: packaging/os/homebrew_cask.py [hunk 1]
```diff
@@ -488,15 +515,27 @@ def _uninstall_casks(self):
...
-            path=dict(required=False),
+            name=dict(
+                aliases=["pkg", "package", "cask"],
+                required=False,
+                type='list',
+            ),
+            path=dict(
+                 default="/usr/local/bin",
+                 required=False,
+            ),
```
Comment (resmo) [inline packaging/os/homebrew_cask.py:527]: Please add `type='path',` to the `path` param.

Comment (indrajitr) [inline packaging/os/homebrew_cask.py:527]: Good one. Done!

**Comment by gregdek** [comment]
Thanks @indrajitr for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by indrajitr** [comment]
ready_for_review

**Comment by indrajitr** [comment]
bot_broken

Travis failure should be ignored for missing param doc fixes.

**Comment by indrajitr** [comment]
@gregdek, how can we move to the next step?

**Comment by gregdek** [comment]
@indrajitr so I'm doing a couple of things here:
1. Marking with bot_broken so we'll stop resetting this issue.
2. Putting back into shipit.

It's not the bot that's broken per se, it's Travis, but the effect is the same.

Now that it's left in shipit, the core team should catch it on its next pass. If it reverts again, please ping me again.

**Comment by gregdek** [comment]
@jctanner does the PR loop also stop on bot_broken? If it doesn't, could you add that?

**Comment by resmo** [comment]
@gregdek @jctanner @sivel issue causing the bot break is https://github.com/sivel/ansible-testing/blob/master/ansible_testing/modules.py#L436

It is not travis per se: there are several cases which let the build fail even if there are valid:
- fixing a missing param doc
- fixing a wrong version_added value
- switching the param name with an alias, which sometimes can make sense.

I would disable this check for now at least.

**Comment by gregdek** [comment]
Thanks @indrajitr for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by indrajitr** [comment]
ready_for_review
... and thus continues the fight with Ansibull 😄

**Comment by resmo** [comment]
LGTM

Shipping it on owner request, broken build can be ignored (issue with version_added checks is addressed)

**Pull request merged**